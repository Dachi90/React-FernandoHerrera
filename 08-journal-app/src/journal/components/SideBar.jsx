import { TurnedInNot } from '@mui/icons-material';
import { Box, Divider, Drawer, Grid, List, ListItem, ListItemButton, ListItemIcon, ListItemText, Toolbar, Typography } from '@mui/material';

export const SideBar = ({ drawerWith = 240 }) => {
	return (
		<Box
			component='nav'
			sx={{ width: { sm: drawerWith }, flexShrink: { sm: 0 } }}
		>
			<Drawer
				variant='permanent' // temporary
				open
				sx={{
					display: { xs: 'block' },
					'& .MuiDrawer-paper': { boxSizing: 'border-box', width: drawerWith },
				}}
			>
				<Toolbar>
					<Typography
						variant='h6'
						noWrap
						component='div'
					>
						Martin Bergesio
					</Typography>
				</Toolbar>
				<Divider />
				<List>
					{['Enero', 'Febrero', 'Marzo', 'Abril'].map((text) => (
						<ListItem
							key={text}
							disablePadding
						>
							<ListItemButton>
								<ListItemIcon>
									<TurnedInNot></TurnedInNot>
								</ListItemIcon>
								<Grid container>
									<ListItemText primary={text} />
									<ListItemText secondary={'Voluptate optio temporibus exercitationem.'} />
								</Grid>
							</ListItemButton>
						</ListItem>
					))}
				</List>
			</Drawer>
		</Box>
	);
};
