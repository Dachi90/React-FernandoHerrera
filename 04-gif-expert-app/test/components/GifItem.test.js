import { render, screen } from '@testing-library/react';
import { GifItem } from '../../src/components/GifItem';

describe('Probando el componente GifItem', () => {
	const title = 'títuloo';
	const url = 'https://probando.com/';

	test('Debe ahcer match con el snapshot', () => {
		const { container } = render(<GifItem title={title} url={url} />);
		expect(container).toMatchSnapshot();
	});

	test('Debe mostrar la imagen con el URL y el ALT indicado', () => {
		render(<GifItem title={title} url={url} />);

		const { src, alt } = screen.getByRole('img');
		expect(src).toBe(url);
		expect(alt).toBe(title);
	});

	test('Debe de mostrar el título en el componente', () => {
		render(<GifItem title={title} url={url} />);
		expect(screen.getByText(title)).toBeTruthy();
	});
});
