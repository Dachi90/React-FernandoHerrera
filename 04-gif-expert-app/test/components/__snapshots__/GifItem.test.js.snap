// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando el componente GifItem Debe ahcer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="títuloo"
      src="url"
    />
    <p>
      títuloo
    </p>
  </div>
</div>
`;
